### PR TITLE
Products

### PR DESCRIPTION
**Pull request by HoolyDuck**
---
Discuss about controllers, db data source and general code

### CODE REVIEW

**Comment by HoolyDuck** [comment]
I will move route strings to enums after this is resolved !

File: server/src/db/migrations/1675371798079-init.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class init1675371798079 implements MigrationInterface {
+    name = 'init1675371798079'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`CREATE TABLE "product_tag" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, CONSTRAINT "PK_1439455c6528caa94fcc8564fda" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "product" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, "imgUrl" character varying NOT NULL, "productTagId" integer, CONSTRAINT "PK_bebc9158e480b949565b4dc7a82" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "role" ("id" SERIAL NOT NULL, "name" character varying NOT NULL, CONSTRAINT "PK_b36bcfe02fc8de3c57a8b2391c2" PRIMARY KEY ("id"))`);
+        await queryRunner.query(`CREATE TABLE "user" ("id" SERIAL NOT NULL, "nickname" character varying NOT NULL, "email" character varying NOT NULL, "password" character varying NOT NULL, CONSTRAINT "PK_cace4a159ff9f2512dd42373760" PRIMARY KEY ("id"))`);
```
Comment (Gr1ngoire) [inline server/src/db/migrations/1675371798079-init.ts]: As I asked, you need to remove this

File: server/src/db/migrations/1675371798079-init.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        await queryRunner.query(`ALTER TABLE "product" ADD CONSTRAINT "FK_4a1df9cbd7e951807d16587e387" FOREIGN KEY ("productTagId") REFERENCES "product_tag"("id") ON DELETE NO ACTION ON UPDATE NO ACTION`);
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "product" DROP CONSTRAINT "FK_4a1df9cbd7e951807d16587e387"`);
+        await queryRunner.query(`DROP TABLE "user"`);
+        await queryRunner.query(`DROP TABLE "role"`);
```
Comment (Gr1ngoire) [inline server/src/db/migrations/1675371798079-init.ts]: And remove this

**Review by Gr1ngoire — CHANGES_REQUESTED** [review]

File: server/src/db/migrations/1675371798079-init.ts (hunk 2 above)
Comment (HoolyDuck) [reply server/src/db/migrations/1675371798079-init.ts]: oops

**Review by Gr1ngoire — APPROVED** [review]

**Pull request merged**